File: 7-kyu/looking-for-a-benefactor/c++/solution.cpp
#include <cmath>
#include <stdexcept>
#include <numeric>
#include <vector>

class NewAverage
{
public:
   static long long newAvg(std::vector<double> &arr, double navg)
   {
       long long ret = std::ceil(navg * (arr.size() + 1) - std::accumulate(arr.begin(), arr.end(), 0.0));
       if (ret <= 0) throw std::logic_error("Logic error");
       return ret;
   }
};

File: 7-kyu/looking-for-a-benefactor/javascript/solution.js
const newAvg = (arr, newavg) => {
  const ret = Math.ceil(newavg * (arr.length + 1) - arr.reduce((acc, nxt) => acc + nxt, 0));
  if (ret <= 0) {
    throw "Error: last donation is non-positive";
  }
  return ret;
};